src/main/java/gov/cms/madie/services/MeasureMapper.java

```diff
@@ -142,82 +152,135 @@ default MeasureGroupingType measureToMeasureGroupingType(QdmMeasure measure) {
       return null;
     }
-
+    Set<CQLDefinition> cqlDefinitions = cqlLookups.getDefinitions();
     // Clauses are listed in the order Populations, Observations, Stratums
     List<ClauseType> clauses = new ArrayList<>();
     if (!CollectionUtils.isEmpty(group.getPopulations())) {
-      clauses.addAll(group.getPopulations().stream().map(this::populationToClauseType).toList());
+      clauses.addAll(
+          group.getPopulations().stream()
+              .map(
+                  population -> {
+                    CQLDefinition cqlDefinition =
+                        getCqlDefinition(population.getDefinition(), cqlDefinitions);
+                    return populationToClauseType(population, cqlDefinition);
+                  })
+              .toList());
     }
     if (!CollectionUtils.isEmpty(group.getMeasureObservations())) {
       clauses.addAll(
-          group.getMeasureObservations().stream().map(this::observationToClauseType).toList());
+          group.getMeasureObservations().stream()
+              .map(
+                  observation -> {
+                    CQLDefinition cqlDefinition =
+                        getCqlDefinition(observation.getDefinition(), cqlDefinitions);
+                    String associatedPopulationUUID = "";
+                    if ("Ratio".equals(group.getScoring())) {
+                      Population associatedPopulation =
+                          group.getPopulations().stream()
+                              .filter(
+                                  population ->
+                                      population.getId().equals(observation.getCriteriaReference()))
+                              .findFirst()
+                              .orElse(null);
+                      associatedPopulationUUID =
+                          associatedPopulation != null ? associatedPopulation.getId() : "";
+                    }
+                    return observationToClauseType(
+                        observation, cqlDefinition, associatedPopulationUUID);
+                  })
+              .toList());
     }
     if (!CollectionUtils.isEmpty(group.getStratifications())) {
       clauses.addAll(
-          group.getStratifications().stream().map(this::stratificationToClauseType).toList());
+          group.getStratifications().stream()
+              .map(
+                  stratification -> {
+                    CQLDefinition cqlDefinition =
+                        getCqlDefinition(stratification.getCqlDefinition(), cqlDefinitions);
+                    return stratificationToClauseType(stratification, cqlDefinition);
+                  })
+              .toList());
     }
     return CollectionUtils.isEmpty(clauses) ? null : clauses;
   }
 
+  default CQLDefinition getCqlDefinition(String definition, Set<CQLDefinition> cqlDefinitions) {
+    return cqlDefinitions.stream()
+        .filter(cqlDefinition -> definition.equals(cqlDefinition.getDefinitionName()))
+        .findFirst()
```

Comment:
we'll have to keep this in mind when we re-enable Ratio w/ 2 IPs..

Reply:
yeah, right

Reply:
or maybe not...I think it would be fine. CQL definitions are unique.